lessons/advanced/behaviours.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: page
+title: Behaviours
+category: advanced
+order: 
+lang: en
+---
+
+We learned about Typespecs in the previous lesson, here we'll learn about how to require a module implement those specifications.  In Elixir, this functionality is referred to as Behaviours.
+
+{% include toc.html %}
+
+## Uses
+
+Sometimes you want modules to share a public API, the solution for this in Elixir is behaviours. Behaviours perform two primary roles:
+
++ Defining a set of function that must be implemented
++ Checking whether that set was actually implemented
+
+Elixir includes a number of behaviors such as GenServer, but in this lesson we'll focus on creating our own instead.
```

Comment:
Here you spell it `behaviors` instead of `behaviours`; we should stick to one spelling. I dont have a preference to the british vs american english style as long as its consistent.